### DISCUSSION TITLE
get resource specific version

### DISCUSSION THREAD

**Discussion by schmurfy** [concourse/concourse#5650] [General]
I am not clear on how to specify the version for `get`step in job, for example if I want to get two resources on the same version, can I do the following?

```json
 "plan": [
    {
       "get": "res1",
       "trigger": true
    },
    {
       "get": "res2",
       "version": "res1"
    }
]
```

And if yes what can actually be passed as version, can it be produced by the output of a task like in this pseudo code ?

```json
 "plan": [
    {
       "get": "res1",
       "trigger": true
    },
    {
      "task": "something",
      "config": {
        "inputs": [
          {"name": "res1"}
        ],
        
        "outputs": [
          {"name": "new_version"}
        ]
      }
    },
    {
       "get": "res2",
       "version": "new_version
    }
]
```

Sorry if I missed something but I read everything I found on this and I can't wrap my head on how it actually works.

**Comment by jamieklassen** [comment]
Can you be more explicit about these resources, perhaps giving a complete sample pipeline config? For example what type of resources are they? The docs give an example with a git resource, where you can use the `version:` field to configure a job to always fetch a static/fixed/constant version of a resource: https://concourse-ci.org/jobs.html#schema.step.get-step.version. This kind of "pinned/hard-coded" workflow is the use case for which the feature was designed.

When you ask 

> what can actually be passed as version

the answer is

> An object with string keys and string values

https://concourse-ci.org/config-basics.html#schema.version, such as might appear in the `stdout` stream of the `check`, `in` or `out` script of a resource: https://concourse-ci.org/implementing-resource-types.html.

What you have in mind sounds like it is subtly more dynamic, and I'd be curious to understand what it means to have the _same_ version of two _different_ resources. Also potentially of interest is: whatever "getting two resources on the same version" means, why are you doing it/what is the purpose of this job?

**Reply by schmurfy** [reply]
I am playing with concourse to try building a pipeline for a golang repository which includes multiple applications, I wanted to provide a cache of dependencies so they are not fetched every time so I tried a few options including the npm cache way and finally found one I like:

For building the master branch everything is fine:
```
[git resource] => [cache task (pull the source and its depdencies and store them in s3 in a folder as tgz)
[s3 resource] => [build] => [release, ...]
```

For PR it also works the same way and works well:
```
[github-pr] => [cache task (uses the pr number as key in s3 and stores the version in the folder)]
[s3-resource] => [build]
```

The problem arise when I want to integrate in this flow updating the github-pr resource since it requires a version of the form: `{ref: xxx, commit: xxx}`, I have the data but not immediately in this form, this is the reason for the second plan in my first post.

I don't actually need the `github-pr` resource to update the pr or at least I hoped, but as it turns out there no exiting cli tool which does that and I am motivated enough yet to make my own cli tool, this is why I wanted to find a way to fetch both my cached folder and the associated github-pr.
